docs/index.md

```diff
@@ -145,9 +140,57 @@ illustrates how to do this:
     	$ unset DOCKER_TLS_VERIFY
       ```
 
+## Discovery (optional)
+
+Instead of providing a comma-separated list of machines, Swarm can be configured to
+dynamically discover machines.
+
+This allows to add and remove machines from the cluster without having to reconfigure and restart
+the Swarm manager.
+
+For this to work, each machine will run a Swarm agent. The agent registers the referenced
+Docker daemon, monitors it, and updates the discovery backend with the machine's status.
+
+Swarm supports a wide range of discovery backends, including the `Docker Hub`, `consul`, `zookeeper` and `etcd`.
+
+The following example uses the Docker Hub based `token` discovery service:
+
+1. Create a Swarm cluster using the `docker` command.
+
+    ```bash
```

Comment:
Might want to check if the Docker docs support this (i.e. the "bash" / language hint), it's not a standard Markdown feature, so it may not work (haven't checked)